src/Databases/DatabaseReplicatedWorker.cpp

```diff
@@ -459,4 +464,9 @@ UInt32 DatabaseReplicatedDDLWorker::getLogPointer() const
     return max_id.load();
 }
 
+UInt64 DatabaseReplicatedDDLWorker::getCurrentInitializationDurationMs() const
+{
+    return initializing.load(std::memory_order_acquire) ? initialization_duration_timer.elapsedMilliseconds() : 0;
```

Comment:
There's still a (theoretical) race condition:
 - thread X checks `initializing` here, it's true, but kernel decides to pause this thread for a few seconds for some reason
 - the DDLWorker thread finishes the initialization successfully, but instantly gets an unexpected error and calls `initializeMainThread` again
 - thread X wakes up and reads `start_ns` inside `initialization_duration_timer.elapsedMilliseconds()` at the same time when the DDLWorker thread resets it in `initialization_duration_timer.restart()`